src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,132 @@
+            } else if ("delete".equalsIgnoreCase(commandType)) {
+               return deleteTask(arguments);
+            } else {
+                throw new InvalidInputException();
+            }
+    }
+
+    private static Command addToDo (String[] arguments) throws EmptyDescriptionException {
+        if (arguments.length == 1) {
+            throw new EmptyDescriptionException();
+        }
+        String toDo = arguments[1];
+        return new AddCommand(new ToDo(toDo, UNDONE_STATUS));
+    }
+
+    private static Command addDeadline (String[] arguments) throws EmptyDescriptionException {
+        if (arguments.length == 1) {
+            throw new EmptyDescriptionException();
+        }
+        String[] inputs = arguments[1].split("/");
+        String description = inputs[0];
+        LocalDate date = LocalDate.parse(extractDateByKeyword("by",inputs[1]));
+        return new AddCommand(new Deadline(description, date, UNDONE_STATUS));
+    }
+
+    private static Command addEvent (String[] arguments) throws EmptyDescriptionException {
+        if (arguments.length == 1) {
+            throw new EmptyDescriptionException();
+        }
+        String[] inputs = arguments[1].split("/");
+        String description = inputs[0];
+        LocalDate date = LocalDate.parse(extractDateByKeyword("at",inputs[1]));
+        return new AddCommand(new Event(description, date, UNDONE_STATUS));
+    }
+
+    private static Command deleteTask (String[] arguments) throws EmptyDescriptionException {
+        if (arguments.length == 1) {
+            throw new EmptyDescriptionException();
+        }
+        int taskNo = Integer.parseInt(arguments[1]);
+        return new DeleteCommand(taskNo);
+    }
+
+    private static Command markTask (String[] arguments) throws EmptyDescriptionException {
+        if (arguments.length == 1) {
+            throw new EmptyDescriptionException();
+        }
+        int taskNo = Integer.parseInt(arguments[1]);
+        return new MarkCommand(taskNo);
+    }
+
+    private static Command unMarkTask (String[] arguments) throws EmptyDescriptionException {
+        if (arguments.length == 1) {
+            throw new EmptyDescriptionException();
+        }
+        int taskNo = Integer.parseInt(arguments[1]);
+        return new UnMarkCommand(taskNo);
+    }
+
+    private static Command listAllTasks ()  {
+        return new ListCommand();
+    }
+
+    private static Command listTasksOnDate (String[] arguments) throws EmptyDescriptionException{
+        if (arguments.length == 1) {
+            throw new EmptyDescriptionException();
+        }
+        LocalDate date = LocalDate.parse(arguments[1]);
+        return new DateCommand(date);
+    }
+
+    private static Command exit ()  {
+        return new ByeCommand();
+    }
+    private static String extractDateByKeyword (String keyword, String text) {
+        String[] args = text.split(keyword);
+        String date = args[1].trim();
+        return date;
+    }
+}
```

Comment:
Nice way of keeping your main method clean by putting all of the relevant functionality in at the bottom in private methods 👍